roxctl/generate/netpol/cta.go

```diff
@@ -12,12 +17,82 @@ func (cmd *generateNetpolCommand) generateNetpol() error {
 		return errors.Wrap(err, "Error synthesizing policies from folder")
 	}
 
+	var yamlPolicies []string
+	var mergedPolicy string
 	for _, netpol := range recommendedNetpols {
 		yamlPolicy, err := networkpolicy.KubernetesNetworkPolicyWrap{NetworkPolicy: netpol}.ToYaml()
 		if err != nil {
-			return errors.Wrap(err, "Error converting YAML into Network Policy")
+			return errors.Wrap(err, "Error converting Network Policy object to YAML")
+		}
+		yamlPolicies = append(yamlPolicies, yamlPolicy)
+		mergedPolicy = fmt.Sprintf("%s---\n%s", mergedPolicy, yamlPolicy)
+	}
+
+	if !cmd.mergePolicies && !cmd.splitPolicies {
+		cmd.printNetpolsToStdout(mergedPolicy)
+		return nil
+	}
+
+	if cmd.mergePolicies {
+		if err := cmd.saveNetpolsToMergedFile(mergedPolicy); err != nil {
+			return errors.Wrapf(err, "Error writing merged Network Policies to file")
+		}
+	}
+
+	if cmd.splitPolicies {
+		if err := cmd.saveNetpolsToFolder(recommendedNetpols); err != nil {
+			return errors.Wrapf(err, "Error writing split Network Policies to folder")
 		}
-		cmd.env.Logger().PrintfLn("---\n\n%s", yamlPolicy)
 	}
+
 	return nil
 }
+
+func (cmd *generateNetpolCommand) printNetpolsToStdout(combinedNetpols string) {
+	cmd.env.Logger().PrintfLn(combinedNetpols)
```

Comment:
`cmd.env.Logger` is not necessarily `stdout`.

Reply:
That's correct, thanks! I've renamed the function to remove the notion of stdout.